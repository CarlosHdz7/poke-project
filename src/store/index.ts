/* eslint-disable no-underscore-dangle */
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../rootReducer';

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(thunk),
    (window as any).__REDUX_DEVTOOLS_EXTENSION__ ? (window as any).__REDUX_DEVTOOLS_EXTENSION__() : (f: any) => f,
  ),
);

export default store;
