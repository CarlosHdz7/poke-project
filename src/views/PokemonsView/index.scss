.cards-container {
  display: flex;
  flex-wrap: wrap;
  align-items: space-between;
  justify-content: center;
  margin: 0 auto;
  width: 100%;
  padding-bottom: 10px;
  box-sizing: border-box;
}
.searchbar-container {
  position: relative;
}

.searchbar-container__icon {
  position: absolute;
  top: 11px;
  left: 22px;
}

.searchbar-container__input {
  box-sizing: border-box;
  width: calc(100% - 20px);
  margin: 10px;
  margin-top: 0px;
  margin-bottom: 20px;
  border: none;
  padding: 10px;
  padding-left: 40px;
  border-radius: 8px;
  font-size: 20px;
}

.main-container {
  padding-bottom: 40px;
}
