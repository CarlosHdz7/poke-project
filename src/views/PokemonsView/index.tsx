import React, { useState } from 'react';
import { useSelector } from 'react-redux';

import { useSearchParams } from 'react-router-dom';
import Loader from '../../components/general/loader';
import Pagination from '../../components/general/Pagination';
import PokemonCard from '../../components/general/PokemonCard';
import useFetchPokemons from '../../hooks/useFetchPokemons';
import IPokemon from '../../interfaces/IPokemon';
import { RootState } from '../../rootReducer';
import './index.scss';

const PokemonsView = () => {
  const [search, setSearch] = useSearchParams();
  const [nameState, setNameState] = useState(search.get('name')?.trim() || '');
  const [pageState, setPageState] = useState(search.get('page')?.trim() || '1');
  const { data: pokemons, loading, error } = useFetchPokemons({ name: nameState, page: pageState });
  const bookmarks: any = useSelector((state: RootState) => state.bookmarks);
  console.log(bookmarks);

  const handleSearch = (value: string) => {
    const pokemonName = value.trim();
    setNameState(pokemonName);
    setSearch(pokemonName ? { name: pokemonName } : {});
  };

  const handlePage = (value: string) => {
    const pageNumber = value.trim();
    setPageState(pageNumber);
    setSearch(pageNumber ? { page: pageNumber } : {});
  };

  return (
    <>
      <h1 className="title">Pokemons</h1>
      <div className="searchbar-container">
        <i className="bi bi-search searchbar-container__icon" />
        <input
          value={nameState}
          type="text"
          className="searchbar-container__input"
          placeholder="Search ..."
          onChange={(e) => handleSearch(e.target.value)}
        />
      </div>

      {!loading && !error && (
        <div className="main-container">
          {!nameState && <Pagination handlePage={handlePage} currentPage={pageState} />}

          <div className="cards-container">
            {pokemons?.map(({ image, name, id, type }: IPokemon) => (
              <PokemonCard key={id} image={image} name={name} id={id} type={type} />
            ))}
          </div>

          {!nameState && <Pagination handlePage={handlePage} currentPage={pageState} />}
        </div>
      )}

      {!pokemons?.length && !loading && 'No results'}

      {loading && <Loader />}

      {error && 'An error has ocurred ...'}
    </>
  );
};

export default PokemonsView;
