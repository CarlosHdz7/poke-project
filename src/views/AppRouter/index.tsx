import React from 'react';
import { Route, Routes } from 'react-router-dom';

import Navbar from 'components/layout/Navbar';
import AboutView from 'views/AboutView';
import BookmarksView from 'views/BookmarksView';
import PokemonDetailView from 'views/PokemonDetailView';
import PokemonsView from 'views/PokemonsView';
import Error404 from 'views/Errors/Error404';
import appRoutes from 'routes';

const AppRouter = () => (
  <>
    <Navbar />
    <div className="container">
      <Routes>
        <Route path={appRoutes.POKEMONS} element={<PokemonsView />} />
        <Route path={appRoutes.POKEMON_DETAIL} element={<PokemonDetailView />} />
        <Route path={appRoutes.BOOKMARKS} element={<BookmarksView />} />
        <Route path={appRoutes.ABOUT} element={<AboutView />} />
        <Route path={appRoutes.ERROR_404} element={<Error404 />} />
        <Route path={appRoutes.HOME} element={<PokemonsView />} />
      </Routes>
    </div>
  </>
);

export default AppRouter;
