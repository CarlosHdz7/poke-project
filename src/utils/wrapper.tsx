/* eslint-disable react/react-in-jsx-scope */
import { ReactElement } from 'react';

import { render } from '@testing-library/react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from 'store';
import AppRouter from 'views/AppRouter';

const renderWithBrowseRouter = (component: ReactElement) => {
  return render(
    <Provider store={store}>
      <BrowserRouter>{component}</BrowserRouter>
    </Provider>,
  );
};

const renderWithMemoryRouter = (path: string[] = ['/']) => {
  return render(
    <Provider store={store}>
      <MemoryRouter initialEntries={path}>
        <AppRouter />
      </MemoryRouter>
    </Provider>,
  );
};

export { renderWithBrowseRouter, renderWithMemoryRouter };
