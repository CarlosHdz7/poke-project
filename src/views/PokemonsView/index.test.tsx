import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import PokemonsView from '.';
import { renderWithMemoryRouter } from 'utils/wrapper';
import { Provider } from 'react-redux';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import store from 'store';
import AppRouter from 'views/AppRouter';

describe('Testing Pokemon list view', () => {
  beforeEach(() => {
    jest.spyOn(console, 'warn').mockImplementation(() => {});
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  it('should renders a list of pokemons initially', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <PokemonsView />
        </BrowserRouter>
      </Provider>,
    );
    await screen.findByText('Bulbasaur');
    await screen.findByText('Ivysaur');
    await screen.findByText('Charmander');
    await screen.findByText('Raticate');
  });

  it('should shows pokemon of page 2', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <PokemonsView />
        </BrowserRouter>
      </Provider>,
    );
    const buttons = await screen.findAllByText(/2/i);

    fireEvent.click(buttons[0]);
    await screen.findByText('Spearow');
  });

  it('should redirect a pokemon details', async () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/']}>
          <PokemonsView />
          <AppRouter />
        </MemoryRouter>
      </Provider>,
    );

    const card = await screen.findByRole('button', {
      name: /Bulbasaur/i,
    });
    fireEvent.click(card);

    expect(await screen.findByText('Bulbasaur')).toBeInTheDocument();
    expect(await screen.findByText('Attack')).toBeInTheDocument();
    expect(await screen.findByText('Defense')).toBeInTheDocument();
  });

  it('should search a pokemon', async () => {
    renderWithMemoryRouter(<PokemonsView />, ['/'], true);

    const searchBar = await screen.findAllByPlaceholderText(/Search/i);

    fireEvent.change(searchBar[0], { target: { value: 'Pikachu' } });
    await new Promise((r) => setTimeout(r, 1000));
    expect(await screen.findByText('Pikachu')).toBeInTheDocument();

    fireEvent.change(searchBar[0], { target: { value: '' } });
    await new Promise((r) => setTimeout(r, 1000));
    expect(await screen.findByText('Bulbasaur')).toBeInTheDocument();
  });

  it('should show no results', async () => {
    renderWithMemoryRouter(<PokemonsView />, ['/'], true);

    const searchBar = await screen.findAllByPlaceholderText(/Search/i);

    fireEvent.change(searchBar[0], { target: { value: 'sfaasf' } });
    await new Promise((r) => setTimeout(r, 1000));
    expect(await screen.findByText(/No results/i)).toBeInTheDocument();
  });
});
