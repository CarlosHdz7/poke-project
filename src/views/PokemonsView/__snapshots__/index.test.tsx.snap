// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pokemon list view should redirect to pokemon details 1`] = `
<div>
  <div
    class="navbar shadow"
  >
    <button
      class="side-button"
      type="button"
    >
      <i
        class="bi bi-list"
      />
    </button>
    <div
      class="navbar-container-options"
    >
      <a
        class="navbar-container-item navbar-container-item__logo"
        href="/"
      >
        <img
          alt=""
          class="navbar-image"
          src="pokeball.png"
        />
        PokeApi
      </a>
      <div>
        <a
          class="navbar-container-item"
          href="/bookmarks"
        >
          Bookmarks
        </a>
        <a
          class="navbar-container-item"
          href="/about"
        >
          About
        </a>
      </div>
    </div>
  </div>
  <nav
    class="side-menu-container"
  >
    <ul
      class="side-menu-items"
    >
      <button
        class="side-menu__item--button side-button-close"
        type="button"
      >
        <i
          class="bi bi-x-lg"
        />
      </button>
      <li
        class="side-menu__item side-bar-text"
      >
        Side Bar Options
      </li>
      <li
        class="side-menu__item"
      >
        <a
          class=""
          href="/"
        >
          <span>
            PokeApi
          </span>
        </a>
      </li>
      <li
        class="side-menu__item"
      >
        <a
          class=""
          href="/bookmarks"
        >
          <span>
            Bookmarks
          </span>
        </a>
      </li>
      <li
        class="side-menu__item"
      >
        <a
          class=""
          href="/about"
        >
          <span>
            About
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="container"
  >
    <div
      class="detail-container shadow"
    >
      <div
        class="detail-container-options"
      >
        <i
          aria-hidden="true"
          class="bi bi-arrow-left-circle-fill"
        />
        <i
          aria-hidden="true"
          class="bi bi-heart"
        />
      </div>
      <h1
        class="detail-container-title"
      >
        Name: 
        Bulbasaur
      </h1>
      <div
        class="inner-detail-container"
      >
        <img
          alt=""
          class="inner-detail-image"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
        />
        <div
          class="types"
        >
          <span
            class="badged grass"
          >
            grass
          </span>
          <span
            class="badged poison"
          >
            poison
          </span>
        </div>
        <div
          class="stats"
        >
          <table
            class="stats__table"
          >
            <tbody>
              <tr>
                <td>
                  Hp
                </td>
                <td
                  class="base-column"
                >
                  45
                </td>
                <td
                  class="bar-column"
                >
                  <div
                    class="bar-container shadow"
                  >
                    <div
                      class="bar-container__bar bar-container__bar--Hp"
                      style="width: 22.5%;"
                    />
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  Attack
                </td>
                <td
                  class="base-column"
                >
                  49
                </td>
                <td
                  class="bar-column"
                >
                  <div
                    class="bar-container shadow"
                  >
                    <div
                      class="bar-container__bar bar-container__bar--Attack"
                      style="width: 24.5%;"
                    />
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  Defense
                </td>
                <td
                  class="base-column"
                >
                  49
                </td>
                <td
                  class="bar-column"
                >
                  <div
                    class="bar-container shadow"
                  >
                    <div
                      class="bar-container__bar bar-container__bar--Defense"
                      style="width: 24.5%;"
                    />
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  Special-Attack
                </td>
                <td
                  class="base-column"
                >
                  65
                </td>
                <td
                  class="bar-column"
                >
                  <div
                    class="bar-container shadow"
                  >
                    <div
                      class="bar-container__bar bar-container__bar--Special-Attack"
                      style="width: 32.5%;"
                    />
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  Special-Defense
                </td>
                <td
                  class="base-column"
                >
                  65
                </td>
                <td
                  class="bar-column"
                >
                  <div
                    class="bar-container shadow"
                  >
                    <div
                      class="bar-container__bar bar-container__bar--Special-Defense"
                      style="width: 32.5%;"
                    />
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  Speed
                </td>
                <td
                  class="base-column"
                >
                  45
                </td>
                <td
                  class="bar-column"
                >
                  <div
                    class="bar-container shadow"
                  >
                    <div
                      class="bar-container__bar bar-container__bar--Speed"
                      style="width: 22.5%;"
                    />
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing Pokemon list view should renders a list of pokemons initially 1`] = `
<div>
  <h1
    class="title"
  >
    Pokemons List
  </h1>
  <div
    class="searchbar-container"
  >
    <i
      class="bi bi-search searchbar-container__icon"
    />
    <input
      class="searchbar-container__input"
      placeholder="Search ..."
      type="text"
    />
  </div>
  <div
    class="main-container"
  >
    <div
      class="pagination"
    >
      <button
        class="pagination__button shadow pagination__button--active "
        type="button"
      >
        1
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        2
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        3
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        4
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        5
      </button>
    </div>
    <div
      class="cards-container"
    >
      <button
        class="card shadow"
        name="bulbasaur"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
          />
        </div>
        <p
          class="card-title"
        >
          Bulbasaur
        </p>
        <p>
          <span
            class="badged grass"
          >
            grass
          </span>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="ivysaur"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png"
          />
        </div>
        <p
          class="card-title"
        >
          Ivysaur
        </p>
        <p>
          <span
            class="badged grass"
          >
            grass
          </span>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="venusaur"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/3.png"
          />
        </div>
        <p
          class="card-title"
        >
          Venusaur
        </p>
        <p>
          <span
            class="badged grass"
          >
            grass
          </span>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="charmander"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/4.png"
          />
        </div>
        <p
          class="card-title"
        >
          Charmander
        </p>
        <p>
          <span
            class="badged fire"
          >
            fire
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="charmeleon"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/5.png"
          />
        </div>
        <p
          class="card-title"
        >
          Charmeleon
        </p>
        <p>
          <span
            class="badged fire"
          >
            fire
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="charizard"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/6.png"
          />
        </div>
        <p
          class="card-title"
        >
          Charizard
        </p>
        <p>
          <span
            class="badged fire"
          >
            fire
          </span>
          <span
            class="badged flying"
          >
            flying
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="squirtle"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/7.png"
          />
        </div>
        <p
          class="card-title"
        >
          Squirtle
        </p>
        <p>
          <span
            class="badged water"
          >
            water
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="wartortle"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/8.png"
          />
        </div>
        <p
          class="card-title"
        >
          Wartortle
        </p>
        <p>
          <span
            class="badged water"
          >
            water
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="blastoise"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/9.png"
          />
        </div>
        <p
          class="card-title"
        >
          Blastoise
        </p>
        <p>
          <span
            class="badged water"
          >
            water
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="caterpie"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/10.png"
          />
        </div>
        <p
          class="card-title"
        >
          Caterpie
        </p>
        <p>
          <span
            class="badged bug"
          >
            bug
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="metapod"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/11.png"
          />
        </div>
        <p
          class="card-title"
        >
          Metapod
        </p>
        <p>
          <span
            class="badged bug"
          >
            bug
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="butterfree"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/12.png"
          />
        </div>
        <p
          class="card-title"
        >
          Butterfree
        </p>
        <p>
          <span
            class="badged bug"
          >
            bug
          </span>
          <span
            class="badged flying"
          >
            flying
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="weedle"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/13.png"
          />
        </div>
        <p
          class="card-title"
        >
          Weedle
        </p>
        <p>
          <span
            class="badged bug"
          >
            bug
          </span>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="kakuna"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/14.png"
          />
        </div>
        <p
          class="card-title"
        >
          Kakuna
        </p>
        <p>
          <span
            class="badged bug"
          >
            bug
          </span>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="beedrill"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/15.png"
          />
        </div>
        <p
          class="card-title"
        >
          Beedrill
        </p>
        <p>
          <span
            class="badged bug"
          >
            bug
          </span>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="pidgey"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/16.png"
          />
        </div>
        <p
          class="card-title"
        >
          Pidgey
        </p>
        <p>
          <span
            class="badged normal"
          >
            normal
          </span>
          <span
            class="badged flying"
          >
            flying
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="pidgeotto"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/17.png"
          />
        </div>
        <p
          class="card-title"
        >
          Pidgeotto
        </p>
        <p>
          <span
            class="badged normal"
          >
            normal
          </span>
          <span
            class="badged flying"
          >
            flying
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="pidgeot"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/18.png"
          />
        </div>
        <p
          class="card-title"
        >
          Pidgeot
        </p>
        <p>
          <span
            class="badged normal"
          >
            normal
          </span>
          <span
            class="badged flying"
          >
            flying
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="rattata"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/19.png"
          />
        </div>
        <p
          class="card-title"
        >
          Rattata
        </p>
        <p>
          <span
            class="badged normal"
          >
            normal
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="raticate"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/20.png"
          />
        </div>
        <p
          class="card-title"
        >
          Raticate
        </p>
        <p>
          <span
            class="badged normal"
          >
            normal
          </span>
        </p>
      </button>
    </div>
    <div
      class="pagination"
    >
      <button
        class="pagination__button shadow pagination__button--active "
        type="button"
      >
        1
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        2
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        3
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        4
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        5
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing Pokemon list view should shows pokemon of page 2 1`] = `
<div>
  <h1
    class="title"
  >
    Pokemons List
  </h1>
  <div
    class="searchbar-container"
  >
    <i
      class="bi bi-search searchbar-container__icon"
    />
    <input
      class="searchbar-container__input"
      placeholder="Search ..."
      type="text"
    />
  </div>
  <div
    class="main-container"
  >
    <div
      class="pagination"
    >
      <button
        class="pagination__button shadow  "
        type="button"
      >
        1
      </button>
      <button
        class="pagination__button shadow pagination__button--active "
        type="button"
      >
        2
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        3
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        4
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        5
      </button>
    </div>
    <div
      class="cards-container"
    >
      <button
        class="card shadow"
        name="spearow"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/21.png"
          />
        </div>
        <p
          class="card-title"
        >
          Spearow
        </p>
        <p>
          <span
            class="badged normal"
          >
            normal
          </span>
          <span
            class="badged flying"
          >
            flying
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="fearow"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/22.png"
          />
        </div>
        <p
          class="card-title"
        >
          Fearow
        </p>
        <p>
          <span
            class="badged normal"
          >
            normal
          </span>
          <span
            class="badged flying"
          >
            flying
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="ekans"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/23.png"
          />
        </div>
        <p
          class="card-title"
        >
          Ekans
        </p>
        <p>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="arbok"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/24.png"
          />
        </div>
        <p
          class="card-title"
        >
          Arbok
        </p>
        <p>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="pikachu"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/25.png"
          />
        </div>
        <p
          class="card-title"
        >
          Pikachu
        </p>
        <p>
          <span
            class="badged electric"
          >
            electric
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="raichu"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/26.png"
          />
        </div>
        <p
          class="card-title"
        >
          Raichu
        </p>
        <p>
          <span
            class="badged electric"
          >
            electric
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="sandshrew"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/27.png"
          />
        </div>
        <p
          class="card-title"
        >
          Sandshrew
        </p>
        <p>
          <span
            class="badged ground"
          >
            ground
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="sandslash"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/28.png"
          />
        </div>
        <p
          class="card-title"
        >
          Sandslash
        </p>
        <p>
          <span
            class="badged ground"
          >
            ground
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="nidoran-f"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/29.png"
          />
        </div>
        <p
          class="card-title"
        >
          Nidoran-f
        </p>
        <p>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="nidorina"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/30.png"
          />
        </div>
        <p
          class="card-title"
        >
          Nidorina
        </p>
        <p>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="nidoqueen"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/31.png"
          />
        </div>
        <p
          class="card-title"
        >
          Nidoqueen
        </p>
        <p>
          <span
            class="badged poison"
          >
            poison
          </span>
          <span
            class="badged ground"
          >
            ground
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="nidoran-m"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/32.png"
          />
        </div>
        <p
          class="card-title"
        >
          Nidoran-m
        </p>
        <p>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="nidorino"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/33.png"
          />
        </div>
        <p
          class="card-title"
        >
          Nidorino
        </p>
        <p>
          <span
            class="badged poison"
          >
            poison
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="nidoking"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/34.png"
          />
        </div>
        <p
          class="card-title"
        >
          Nidoking
        </p>
        <p>
          <span
            class="badged poison"
          >
            poison
          </span>
          <span
            class="badged ground"
          >
            ground
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="clefairy"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/35.png"
          />
        </div>
        <p
          class="card-title"
        >
          Clefairy
        </p>
        <p>
          <span
            class="badged fairy"
          >
            fairy
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="clefable"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/36.png"
          />
        </div>
        <p
          class="card-title"
        >
          Clefable
        </p>
        <p>
          <span
            class="badged fairy"
          >
            fairy
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="vulpix"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/37.png"
          />
        </div>
        <p
          class="card-title"
        >
          Vulpix
        </p>
        <p>
          <span
            class="badged fire"
          >
            fire
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="ninetales"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/38.png"
          />
        </div>
        <p
          class="card-title"
        >
          Ninetales
        </p>
        <p>
          <span
            class="badged fire"
          >
            fire
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="jigglypuff"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/39.png"
          />
        </div>
        <p
          class="card-title"
        >
          Jigglypuff
        </p>
        <p>
          <span
            class="badged normal"
          >
            normal
          </span>
          <span
            class="badged fairy"
          >
            fairy
          </span>
        </p>
      </button>
      <button
        class="card shadow"
        name="wigglytuff"
        type="button"
      >
        <div
          class="img-container"
        >
          <img
            alt=""
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/40.png"
          />
        </div>
        <p
          class="card-title"
        >
          Wigglytuff
        </p>
        <p>
          <span
            class="badged normal"
          >
            normal
          </span>
          <span
            class="badged fairy"
          >
            fairy
          </span>
        </p>
      </button>
    </div>
    <div
      class="pagination"
    >
      <button
        class="pagination__button shadow  "
        type="button"
      >
        1
      </button>
      <button
        class="pagination__button shadow pagination__button--active "
        type="button"
      >
        2
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        3
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        4
      </button>
      <button
        class="pagination__button shadow  "
        type="button"
      >
        5
      </button>
    </div>
  </div>
</div>
`;
